### ISSUE TITLE
ActivateJobsCommandImpl throws NullPointerException when ZeebeClientProperties.getDefaultJobWorkerName is null

### ISSUE THREAD

**Issue by sbuettner** [camunda/zeebe#14176]
**Describe the bug**

The constructor of the `ActivateJobsCommandImpl` is trying to set the `workerName` during construction and expects the `config.getDefaultJobWorkerName())` to be non null. This leads to a `NullPointerException` as the underlying builder doesnt support a null value. This is an issue as clients are able to set the `workerName` using the builder pattern after constructing the command.

```
public ActivateJobsCommandImpl(
      final GatewayStub asyncStub,
      final ZeebeClientConfiguration config,
      final JsonMapper jsonMapper,
      final Predicate<Throwable> retryPredicate) {
    this.asyncStub = asyncStub;
    this.jsonMapper = jsonMapper;
    this.retryPredicate = retryPredicate;
    builder = ActivateJobsRequest.newBuilder();
    requestTimeout(config.getDefaultRequestTimeout());
    timeout(config.getDefaultJobTimeout());
    workerName(config.getDefaultJobWorkerName());
  }
```

**To Reproduce**

Create an implementation of the `ZeebeClientProperties` that returns `null` for the `getDefaultJobWorkerName`

**Expected behavior**

The builder should fallback to the default worker name only if it wasnt set by a client when constructing the final command.

**Log/Stacktrace**

```
java.lang.NullPointerException: null
	at io.camunda.zeebe.gateway.protocol.GatewayOuterClass$ActivateJobsRequest$Builder.setWorker(GatewayOuterClass.java:2169)
	at io.camunda.zeebe.client.impl.command.ActivateJobsCommandImpl.workerName(ActivateJobsCommandImpl.java:83)
	at io.camunda.zeebe.client.impl.command.ActivateJobsCommandImpl.<init>(ActivateJobsCommandImpl.java:60)
```

**Environment:**
- Zeebe Version:8.3.0-alpha4

Looks like it happens as the command builder is used instead of the previously used request builder since: https://github.com/camunda/zeebe/commit/359a402b5bbee7247749385a458c5b3f3aba7e78

**Comment by korthout** [comment]
@npepinpe This sounds like a regression introduced with #12888. Could you please have a look?

**Comment by npepinpe** [comment]
We had a look together, and AFAIK this is the same behavior as ever. We were always calling `workerName` in the constructor of the command, and it makes sense to me to follow Simon's suggestion and only apply defaults in the build method, not in the constructor.

Happy to have a look if someone can point to me where the regression comes from

**Comment by npepinpe** [comment]
> Looks like it happens as the command builder is used instead of the previously used request builder since: https://github.com/camunda/zeebe/commit/359a402b5bbee7247749385a458c5b3f3aba7e78

@sbuettner - can you point out to me where the change from request to command builder happened? From the commit, the code is the same - we extracted an interface out of a class, but the implementation was not changed.

For example, in the "old" implementation:

```java
    final ActivateJobsCommandStep3 activateCommand =
        jobClient
            .newActivateJobsCommand()
            .jobType(jobType)
            .maxJobsToActivate(maxJobsToActivate)
            .timeout(timeout)
            .workerName(workerName);
```

And it was simply moved in the "new" one, but remains the same:

```java
    final ActivateJobsCommandStep3 activateCommand =
        jobClient
            .newActivateJobsCommand()
            .jobType(jobType)
            .maxJobsToActivate(maxJobsToActivate)
            .timeout(timeout)
            .workerName(workerName);
```

I'm also not sure what you mean by request vs command builder :thinking:

**Comment by sbuettner** [comment]
@npepinpe It looks like it was changed here: https://github.com/camunda/zeebe/commit/11c548dd4d46f62119a600da0dc1abad7231c417#diff-7101968ae21254c317a1606e4a1969ec3864b538403541ecc7161351b2bd38ddL103

**Comment by npepinpe** [comment]
Good catch! Seems unrelated to job push, but it is indeed a regression in that sense for 8.3.0. I'll defer to the ZPA team on what the priority for it is.

**Comment by korthout** [comment]
Thanks @npepinpe and @sbuettner for the extra input.

- We'll need to fix this before 8.3 as it regressed recently.
- We consider this mid severity, as a workaround is available (don't set null as the default job worker name)
- We wonder how quickly a user would run into this, do you have to set `null` explicitly, or does it also occur when the client doesn't define any default?
- This would change the impact dramatically, and so our priority
- We want to timebox whether that is the case
- It might be worth it to fix immediately then (but please timebox it)
